src/main.rs

```diff
@@ -530,6 +567,191 @@ async fn listen_tcp(args: ListenTcpArgs) -> anyhow::Result<()> {
     Ok(())
 }
 
+#[cfg(unix)]
+/// Listen on a magicsocket and forward incoming connections to a Unix socket.
+async fn listen_unix(args: ListenUnixArgs) -> anyhow::Result<()> {
+    let socket_path = args.socket_path.clone();
+    let secret_key = get_or_create_secret()?;
+    let endpoint = create_endpoint(secret_key, &args.common, vec![args.common.alpn()?]).await?;
+    // wait for the endpoint to figure out its address before making a ticket
+    endpoint.home_relay().initialized().await?;
+    let node_addr = endpoint.node_addr().await?;
+    let mut short = node_addr.clone();
+    let ticket = NodeTicket::new(node_addr);
+    short.direct_addresses.clear();
+    let short = NodeTicket::new(short);
+
+    // print the ticket on stderr so it doesn't interfere with the data itself
+    //
+    // note that the tests rely on the ticket being the last thing printed
+    eprintln!(
+        "Forwarding incoming requests to '{}'.",
+        socket_path.display()
+    );
+    eprintln!("To connect, use e.g.:");
+    eprintln!("dumbpipe connect-unix --socket-path /path/to/client.sock {ticket}");
+    eprintln!("dumbpipe connect-tcp --addr 127.0.0.1:8080 {ticket}");
+    if args.common.verbose > 0 {
+        eprintln!("or:\ndumbpipe connect-unix --socket-path /path/to/client.sock {short}");
+        eprintln!("dumbpipe connect-tcp --addr 127.0.0.1:8080 {short}");
+    }
+    tracing::info!("node id is {}", ticket.node_addr().node_id);
+    tracing::info!("derp url is {:?}", ticket.node_addr().relay_url);
+
+    // handle a new incoming connection on the magic endpoint
+    async fn handle_magic_accept(
```

Comment:
I think we should remove magic terminology from this, since we no longer refer to the endpoint as a magic endpoint.

Reply:
I have been thinking magicpipe would be a better name than dumbpipe for this utility though. It really feels like magic!